### PR TITLE
Mirror of PR on ega-archive

### PR DESCRIPTION
**Pull request by viklund**
This is only so we can get the integration tests running.

---
This change is [Reviewable](https://reviewable.io/reviews/nbisweden/localega/374)

### CODE REVIEW

File: lega/utils/errors.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    file_id = data.get('file_id', None)  # should be there
+    if file_id:
+        set_error(file_id, cause, from_user)
+    LOG.debug('Catching error on file id: %s', file_id)
+    if from_user:  # Send to CentralEGA
+        org_msg = data.pop('org_msg', None)  # should be there
+        org_msg['reason'] = str(cause)  # str = Informal
+        LOG.info('Sending user error to local broker: %s', org_msg)
+        publish(org_msg, 'cega', 'files.error', correlation_id)
```
Comment (blankdots) [inline lega/utils/errors.py:63]: where does this `correlation_id` come from, don't see it defined ?

File: lega/notifications.py [hunk 2]
```diff
@@ -103,24 +98,19 @@ def connection_lost(self, exc):
...
+    server = loop.run_until_complete(loop.create_server(Forwarder, host, port))
 
     # Serve requests until Ctrl+C is pressed
     LOG.info('Serving on %s', host)
     try:
         loop.run_forever()
     except KeyboardInterrupt:
         LOG.info('Server interrupted')
-        broker.close()
+        amqp.close()
```
Comment (blankdots) [inline lega/notifications.py]: where is `amqp` defined, maybe miss ?

File: lega/outgest.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+def main():
+
+    host = CONF.get_value('DEFAULT', 'host')
+    port = CONF.get_value('DEFAULT', 'port', conv=int)
+
+    sslcontext = None
+    if CONF.get_value('DEFAULT', 'enable_ssl', conv=bool, default=True):
+        ssl_certfile = Path(CONF.get_value('DEFAULT', 'ssl_certfile')).expanduser()
+        ssl_keyfile = Path(CONF.get_value('DEFAULT', 'ssl_keyfile')).expanduser()
+        LOG.debug('Certfile: %s', ssl_certfile, extra={'correlation_id': correlation_id})
```
Comment (blankdots) [inline lega/outgest.py]: where does the `correlation_id` come from ? also in the line below

**Review by blankdots — CHANGES_REQUESTED** [review]
first round of review: some misses ?

**Review by jbygdell — CHANGES_REQUESTED** [review]
Rebase and split.
A PR should not contain merge commits from other PRs.

**Review by jbygdell — CHANGES_REQUESTED** [review]
Rebase and split.
A PR should not contain merge commits from other PRs.

**Review by jbygdell — CHANGES_REQUESTED** [review]
Format the commit message properly.
Descriptive title, < 80 characters long terminated by a new line.
Comprehensive description of the commit that clearly specifies what is changed and why.

**Review by jbygdell — CHANGES_REQUESTED** [review]
Format the commit message properly.
Descriptive title, < 80 characters long terminated by a new line.
Comprehensive description of the commit that clearly specifies what is changed and why.

**Review by jbygdell — CHANGES_REQUESTED** [review]
Format the commit message properly.
Descriptive title, < 80 characters long terminated by a new line.
Comprehensive description of the commit that clearly specifies what is changed and why.

**Review by jbygdell — CHANGES_REQUESTED** [review]
47fe5dbb5
Format the commit message properly.
Descriptive title, < 80 characters long terminated by a new line.
Comprehensive description of the commit that clearly specifies what is changed and why.

**Review by jbygdell — CHANGES_REQUESTED** [review]
00ccc14
Rebase and split.
A PR should not contain merge commits from other PRs.

**Review by jbygdell — CHANGES_REQUESTED** [review]
This commit contains other changes that does not match the commit message,
these should be split out to a separate commit to minimise confusion.

File: lega/utils/errors.py (hunk 1 above)
Comment (silverdaz) [reply lega/utils/errors.py:63]: It was defined as an argument, and I added to the context outside the handle_error function. I thought I could remove the argument, my bad, it was a miss. I re-added it.

File: lega/notifications.py (hunk 2 above)
Comment (silverdaz) [reply lega/notifications.py]: Yes, a miss.

File: deploy/images/README.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ In the current folder, type `make` and the images are created in order.
 
 It takes some time.
 
-Later on, if the `crg/ega-*` do need to be recreated, one can type `make`.
```
Comment (jbygdell) [inline deploy/images/README.md]: This commit contains other changes that does not match the commit message,
these should be split out to a separate commit to minimise confusion.

**Review by jbygdell — CHANGES_REQUESTED** [review]

**Pull request closed without merging**